pdbfixer/tests/test_charge_and_solvate.py

```diff
@@ -0,0 +1,57 @@
+import pytest
+from pdbfixer import PDBFixer
+import openmm.unit
+
+
+@pytest.mark.parametrize(
+    "pdbCode,soluteCharge",
+    [
+        ("1PO0", -21),
+        ("1A11", 1),
+        ("110D", -5),
+        ("1CNR", 0),
+        ("1ESD", -21),
+    ],
+)
+def test_charge_and_solvate(pdbCode, soluteCharge):
+    """
+    Test that addSolvent successfully neutralises systems
+
+    Parameters
+    ----------
+    pdbCode : str
+        The PDB ID to test
+    soluteCharge : int
+        The formal charge of the solute - should equal the number of chloride
+        ions minus the number of sodium ions in the neutralised system. Note
+        that this may include ions from the original PDB entry.
+    """
+    fixer = PDBFixer(pdbid=pdbCode)
+    fixer.findMissingResidues()
+
+    chains_to_cap = {chain for chain, resi in fixer.missingResidues}
+    for chainidx in chains_to_cap:
+        chain = [*fixer.topology.chains()][chainidx]
+        last_resi = len([*chain.residues()])
+        # Capping with GLY because ACE/NME currently breaks addMissingHydrogens
+        # Adding a cap keeps the protein compact and the addSolvent call quick
```

Comment:
In what way does it break it?

Reply:
I was running into an issue where if I unconditionally capped the ends of a chain, chains that didn't have missing terminal loops would wind up with unprotonated caps. I mentioned it briefly in #299. I ended up attributing this to a bug in my own code and fixed it by only capping terminals with missing loops, and I've transferred that code over to this PR just now, but I can dig into it further if you like.